rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -309,4 +300,72 @@ void test_extractRedeemScriptFromInput_migration_multiple_inputs() {
             Assertions.assertEquals(expectedRedeemScript, redeemScript.get());
         }
     }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionDoesNotHaveInputs_shouldReturnSameTransaction() {
+        // arrange
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        Sha256Hash transactionHashBeforeRemovingSignatures = transaction.getHash();
+
+        // act
+        BitcoinUtils.removeSignaturesFromTransactionWithInputsWithP2shMultiSigInputScript(transaction);
+
+        // assert
+        Assertions.assertEquals(transactionHashBeforeRemovingSignatures, transaction.getHash());
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionIsSegwit_shouldThrowIllegalArgumentException() {
+        // arrange
+        BtcTransaction transaction = new BtcTransaction(btcMainnetParams);
+        transaction.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        TransactionWitness transactionWitness = new TransactionWitness(1);
+        transactionWitness.setPush(0, new byte[]{});
+        transaction.setWitness(0, transactionWitness);
+
+        // act & assert
+        IllegalArgumentException exception = Assertions.assertThrows(IllegalArgumentException.class,
+            () -> BitcoinUtils.removeSignaturesFromTransactionWithInputsWithP2shMultiSigInputScript(transaction));
+        Assertions.assertEquals("Removing signatures from SegWit transactions is not allowed.", exception.getMessage());
+    }
+
+    @Test
+    void removeSignaturesFromTransaction_whenTransactionInputsDoNotHaveP2shMultiSigInputScript_shouldThrowIllegalArgumentException() {
```

Comment:
a transaction with a mix of p2sh and p2pkh inputs?

Reply:
updated